src/Psl/Arr/flat_map.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Arr;
+
+use Psl\Type;
+
+/**
+ * @psalm-template Tk of array-key
+ * @psalm-template Tv
+ * @psalm-template T
+ *
+ * @psalm-param iterable<Tk, Tv>                $iterable Iterable to be mapped over
+ * @psalm-param (callable(Tv): (T|iterable<T>)) $mapper
```

Comment:
```suggestion
 * @psalm-param (callable(Tv): iterable<T>) $mapper
```

The issue is if T itself is an array or an iterable, we can't really determine if the return value is a single entry or a list.

Reply:
I wanted it to be close to the JS version of flat map, but they have separate syntax for arrays and objects.
A use-case for flatmaps could be this as well:

```php
flat_map(
   [
       fn () => $node1,
       fn () => [$node2, $node3],
   ],
   fn ($builder) => $builder()
)
```

Imagine the initial list of items are composable builders that you'dd also want to use seperately. In that case you might not want to force them all to return iterables. But you do want a node list as result.